otwcode/otwarchive
# AO3-6026 Move collections to Elasticsearch

# Pull Request Checklist

  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6026

## Purpose

What does this PR do?
Creates models for creating and querying elasticsearch indexes for collections.
Modifies collections indexes to use ES to query collections, instead of sql

## Testing Instructions

How can the Archive's QA team verify that this is working as you intended?
Run the command to reindex all collection `CollectionIndexer.index_all` to see these results in the app.
Also, run tests for collection search models and collections controller

(If you have a Jira account with access, please update or comment on the issue
with any new or missing testing instructions.)

## References


Are there other relevant issues/pull requests/mailing list discussions?

## Credit

*What name do you want us to use to credit your work in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?*

(If you have a Jira account, please include the same name in the "Full name"
field on your Jira profile, so we can assign you the issues you're working on.)

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?*

(If you do not fill in this section, we will use your GitHub account name and
they/them pronouns to avoid making assumptions about your gender.)
